git/index/typ.py

```diff
@@ -48,10 +50,10 @@ def __init__(self, paths: Sequence[PathLike]) -> None:
         """
         self.paths = paths
 
-    def __call__(self, stage_blob: Blob) -> bool:
-        path = stage_blob[1].path
+    def __call__(self, stage_blob: Tuple[StageType, Blob]) -> bool:
+        path: str = str(stage_blob[1].path)
         for p in self.paths:
-            if path.startswith(p):
+            if path.startswith(str(p)):
```

Comment:
There has to be a way to do this without `str(…)`, please see above for the reason.